src/shell/shell_cmds.cpp

```diff
@@ -1906,109 +1907,92 @@ void DOS_Shell::CMD_LOADHIGH(char *args){
-	if (args && *args ) {
-		StripSpaces(args);
-		size_t argslen = strlen(args);
-		if (argslen > 1 && args[0] == '"' && args[argslen-1] == '"') {
-			args[argslen-1] = 0; //Remove quotes
-			args++;
-		}
-		WriteOut(args);
-	}
-	/* Show question prompt of the form [a,b]? where a b are the choice values */
-	if (!optN) {
-		if (args && *args) WriteOut(" ");
-		WriteOut("[");
-		size_t len = strlen(rem);
-		for (size_t t = 1; t < len; t++) {
-			WriteOut("%c,",rem[t-1]);
-		}
-		WriteOut("%c]?",rem[len-1]);
-	}
 
-	std::vector<std::string> sequence;
-	bool in_char = optS ? (strchr(rem, wait_char) != nullptr)
-	                    : (strchr(rem, toupper(wait_char)) ||
-	                       strchr(rem, tolower(wait_char)));
-	bool auto_type = wait_char && *rem && in_char && wait_sec > 0;
-	if (auto_type) {
-		sequence.emplace_back(std::string(1, tolower(wait_char)));
-		MAPPER_AutoType(sequence, wait_sec * 1000, 500);
-	}
-	uint16_t n = 1;
-	do {
+	// Parse "/n"; does the user want to see the choices or not?
+	const bool wants_quiet = ScanCMDBool(args, "N");
+
+	// Parse "/s"; does the user want case-sensitive choices?
+	const bool wants_case_sensitive = ScanCMDBool(args, "S");
+
+	// Prepare the command line for use with regular expressions
+	assert(args);
+	std::smatch match;
+	std::string cmdline = args;
+
+	// Parse "/c[:]abc"; has the user provided custom choices?
+	static const std::regex re_choices("/c[:]?([a-zA-Z]+)");
+	const auto has_choices = std::regex_search(cmdline, match, re_choices);
+	auto choices = has_choices ? match[1].str() : "yn";
+	cmdline = std::regex_replace(cmdline, re_choices, "");
+
+	// Parse "/t[:]c,nn"; has a timeout default been provided?
+	static const std::regex re_timeout("/t[:]?([a-zA-Z]),(\\d+)");
+	auto has_default = std::regex_search(cmdline, match, re_timeout);
+	char default_choice = has_default ? match[1].str()[0] : '\0';
+	const auto default_wait_s = has_default ? std::stoi(match[2]) : 0;
+	cmdline = std::regex_replace(cmdline, re_timeout, "");
+
+	// If the user doesn't care about case, convert to upper-case
+	if (!wants_case_sensitive) {
+		upcase(choices);
+		default_choice = static_cast<char>(toupper(default_choice));
+	}
+
+	// Parse the remaining text message(s) and strip whitespace and quotes
+	static const std::regex re_trim("^[\"\\s]+|[\"\\s]+$");
+	const auto message = std::regex_replace(cmdline, re_trim, "");
+	if (message.length())
+		WriteOut(message.c_str());
+
+	// Show question prompt of the form [a,b]? where a b are the choice values
+	if (!wants_quiet) {
+		WriteOut(message.length() ? " [" : "[");
+		for (size_t i = 0; i < choices.length() - 1; i++)
+			WriteOut("%c,", choices[i]);
+		WriteOut("%c]?", choices.back());
+	}
+
+	// If a default was given, is it in the choices and the timeout valid?
+	const auto use_auto_type = has_default &&
+	                           choices.find(default_choice) != std::string::npos &&
```

Comment:
We do have that `contains()` template in `support.h`...

Reply:
Using it now - much easier on the eyes; thanks for the reminder (I had forgot that it would work w/ strings too ).